text/0000-angle-bracket-invocation.md

```diff
@@ -204,36 +204,145 @@ Because `Option` is the name of a local variable (block param), the `<Option>`
+names without a path segment, regarless of whether the name actually collide
+with a known HTML tag – human readers of an Ember template should be able to
+safely assume lowercase tags refer to HTML.
+
+Eventually, we might want to provide stronger guidance with via the linter. For
+example, we may want to recommend capitalizing invokable local variables, as in
+`<F.Input />`. We will let the community experiment and coalesce around these
+conventions before recommending them by default.
+
+Finally, there are two exceptions to the general rule where certain technically
+valid Handlebars path expressions are not supported for dynamic invocations:
+
+* Implicit `this` lookups (a.k.a. "property fallback" in [RFC #308](https://github.com/emberjs/rfcs/pull/308)
+* Slash lookups
+
+First, while `{{foo}}` or `{{Foo}}` can normally refer to `{{this.foo}}` or
+`{{this.Foo}}` normally, allowing this implicitly lookup will mean _any_ tag
+in the template (i.e. `<foo />` or `<Foo />`) can possibly refer to a property
+on the current `this` context.
+
+This ambiguity is highly undesirable for both human readers and the compiler,
+therefore implicitly `this` lookup is not allowed in angle bracket invocations.
+This explicit form, `<this.foo />` and `<this.Foo />` is required.
+
+This requirement aligns well with [RFC #308](https://github.com/emberjs/rfcs/pull/308)
+and the current curly invocation semantics, due to the ["dot rule"](https://github.com/emberjs/rfcs/blob/master/text/0064-contextual-component-lookup.md#component-helper-shorthand)
+that requires a dot in the path. Note that this is actually more restrictive
+than the proposed angle bracket invocation semantics, since it is not possible
+to invoke a local variable without a dot:
+
+```hbs
+{{#super-select selected={{this.user.country}} as |option|>
+  {{#each this.availableCountries as |country|}}
+    {{!-- this is not legal today, since `option` does not contain a dot --}}
+    {{#option value=country}}{{country.name}}{{/option}}
+  {{/each}}
+{{/super-select}}
+```
+
+We propose to relax that rule to match the proposed angle bracket invocation
+semantics (i.e. allowing local variables without a dot, as well as `@names`,
+but disallowing implicit `this` lookup).
+
+Second, while Handlebars technically allows `{{foo/bar}}` as an equivilant
+alternative to the `{{foo.bar}}` path lookup (and therefore `foo/bar` is
+technically a valid Handlebars path expression), it will not be supported in
+angle bracket invocation. This is both because the `/` conflicts with the HTML
+closing tag syntax, and the fact that Ember overrides that syntax with a
+different semantic.
+
+In today's semantics, `{{foo/bar}}` does not try to lookup `this.foo.bar` and
+invoke it as a component. Instead, it is used as a filesystem scoping syntax.
+Since this feature will be rendered unnecessary with [Module Unification](https://github.com/emberjs/rfcs/blob/master/text/0143-module-unification.md),
```

Comment:
Disagree that this feature is unnecessary. 📁 are the most natural way to organize components and are widely used in large applications. None of the alternatives presented in module unification are intuitive or easily transitioned to.